src/lib/sidenav/drawer.ts

```diff
@@ -432,16 +449,27 @@ export class MatDrawerContainer implements AfterContentInit, OnDestroy {
   /** Emits when the component is destroyed. */
   private _destroyed = new Subject<void>();
 
+  /** Cached margins used to verify that the values have changed. */
+  private _margins = {left: 0, right: 0};
+
+  /** Used for debouncing reflows. */
+  private _debouncer = new Subject<void>();
+
   _contentMargins = new Subject<{left: number, right: number}>();
 
-  constructor(@Optional() private _dir: Directionality, private _element: ElementRef,
-              private _renderer: Renderer2, private _ngZone: NgZone,
+  constructor(@Optional() private _dir: Directionality,
+              @Inject(MAT_DRAWER_DEFAULT_AUTOSIZE) defaultAutosize: boolean,
```

Comment:
If we move this to the end of the parameter list and make it optional it won't be breaking for users who are extending this class